contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -451,13 +461,16 @@ contract SmartAccount is
     }
 
     // Extra Utils 
-    function transfer(address payable dest, uint amount) external nonReentrant onlyOwner {
-        require(dest != address(0), "this action will burn your funds");
-        (bool success,) = dest.call{value:amount}("");
-        require(success,"transfer failed");
+    function transfer(address payable dest, uint amount) external payable nonReentrant onlyOwner {
+        if(dest == address(0)) revert TransferToZeroAddressAttempt();
+        bool success;
+        assembly {
+            success := call(gas(), dest, amount, 0, 0, 0, 0)
+        }
+        if (!success) revert TokenTransferFailed(address(0), dest, amount);
     }
 
-    function pullTokens(address token, address dest, uint256 amount) external onlyOwner {
+    function pullTokens(address token, address dest, uint256 amount) external payable onlyOwner {
```

Comment:
nice. should we make all access controlled methods payable?

Reply:
generally we can make payable all the methods that will less probably be called by a user that can accidentally send value

Reply:
hmm right, but the payable modifier does save gas right? or it only applies to constructor?